### DISCUSSION TITLE
Steps to upgrade TimeBase server

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#263] [Q&A]
When ever we upgrade timebase server we lost data as well as streams.
So can you please help us with steps to be taken for upgrading?

**Comment by ankitsoin** [comment]
Also, currently we are using python client 6.0.4 and we have to upgrade it to 6.0.5. 
May I know if I take backup of all existing spaces which were loaded by 6.0.4 version will they work for 6.0.5 version as well?
(Our reason to upgrade version is  to use of spaces while fetching data using python)

**Reply by alex-karpovich** [reply]
@ankitsoin correct, All spaces will work, but I suggest to use 6.1.21 version

**Comment by ankitsoin** [comment]
> When ever we upgrade timebase server we lost data as well as streams. So can you please help us with steps to be taken for upgrading?

do we have any steps for upgrading?

**Reply by ankitsoin** [reply]
Hi @alex-karpovich,

In Community version, while upgrading the TB - by installing Jar file timebase-linux-installer.jar it ask for path so we provide same path where existing myPath/_timebase_ is installed.
So it just overwrite the existing _timebase_ folder(containing sub directories like auto-install.xml, lib, dumps, logs, temp, **timebase**, Uninstaller, web, work etc)

This way we loose data in subdirectory timebase(highlighted in the bold above)  which conatins all stream data.

Is this an expected behaviour while upgrade? that we need to restore all streams data manually post every upgrade?

**Reply by ankitsoin** [reply]
Hi @alex-karpovich, 
Data is getting dropped after every upgrade(Community version) even there is no open writer.

To test the same, I have loaded the data in couple of spaces in each of the 4 streams. Then upgraded the TB. 
After the upgrade, all existing data(in spaces) are present but 2 spaces which I have loaded for testing got dropped in all 4 streams.

Can you please let us know is the expected behavior? Will this behavior be same in enterprise edition as well?

**Reply by alex-karpovich** [reply]
Next approach in important:

1. you install TimeBase Server in lunix under like `opts/timebase` folder
2. you create `timebase-home` which will cotains your data in **SEPARATE** folder, like `/home/<your-username>/timebase-home`
3. start TimeBase using args: `-home /home/<your-username>/timebase-home`

In this case after reinstalling product, your data (in timebase-home) won't be lost
